text/0000-CNB-user-research-2021.md

```diff
@@ -0,0 +1,85 @@
+# Meta
+[meta]: #meta
+- Name: CNB User Research 2021
+- Start Date: 2020-12-10
+- Author(s): sampeinado, speinado@vmware.com
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Background
+The purpose of this study is to understand the major pain points buildpack authors and app developers experience in using Cloud Native Buildpacks and how these issues slow the adoption of CNBs as an emerging standard in containerization. VMware has donated funding for the recruitment and compensation of 15-20 interview subjects to help the CNB project grow. 
+ 
+Given our contributors’ strong understanding of buildpacks and their spec, the project is at risk of a blindspot towards novice buildpack authors who want to create value quickly. The CNB spec enables more modularity than the v2 spec, but its API is also significantly more complex. For our ecosystem of buildpacks to grow, and therefore the number of app types and use cases buildpacks support to grow, we need to develop a better understanding of where buildpack authors struggle and how we can support them. 
+ 
+In addition, we want to better understand why app developers switch to CNBs and what challenges they face when doing so. As a project, we have clear assumptions about why CNBs are valuable relative to other containerization tools, but we don’t know which reasons are most important to which users. Without this clarity, our ability to market the project’s strengths and prioritize the most valuable features is limited. 
+
+# Objectives
+<ol>
+<li>Understand how buildpack authors currently go about writing buildpacks and customizing builds and what goals they are trying to accomplish by doing so</li>
+<ol>
+<li>Evaluate pain points buildpack authors are experiencing </li>
+<li>Evaluate the relative usability of proposals like <a href="https://github.com/sclevine/packfile">packfile</a> or <a href="https://github.com/kr/heroku-buildpack-inline">inline buildpacks</a> </li>
+</ol>
+<li>Define the different segments of app developers using buildpacks, their goals, challenges, and motivations to switch to buildpacks from other tools</li>
+<ol>
+<li>Identify a tipping point at which users can be considered “converted”</li>
+<li>Enable us to become more strategic in our communication in conference talks, blog posts, and working group </li>
+</ol>
+<li>Provide survey responses to key topics the project is currently working on</li>
+<ol>
+<li>How and where they find information today</li>
+<li>Which platforms they use CNBs on</li>
+<li>Expectations around inner loop development</li>
+</ol>
+<li>Anti-goals</li>
+<ol>
+<li>Who’s using what buildpacks brand</li>
+<li>Usability of the `pack` CLI, except as it relates to buildpack authorship</li>
```

Comment:
Given that:
1. One of the goals is to look at app developers and their motivation for trying CNB and any pain points they might experience.
2.  `pack` is the entrypoint for many new CNB users (this assumption might need validation)

Aspects of the usability of the `pack` CLI from the app developer's perspective might be relevant?

Reply:
I agree its hard to disentangle, but all im trying to say here is that the intent is not to improve any particular platfrom implementation per se, but rather the overall buildpacks experience, no matter the platform. this would obviously have implications for the `pack` experience, but the intention is not to be `pack` specific, as much as we can (ie we want the findings to benefit all platform implementations as much as possible)

Reply:
Also, I think at this point the majority of CNB users are using it for the first time through Spring Boot, not `pack`, though I don't have the numbers to prove this